wordpress-mobile/WordPress-iOS
# [Domain Management] Display transfer domains webview from domains search screen

Fixes #21782

## Description

This adds a button in "Register Domain" screen that navigates to "Transfer Domains" web view when tapped.

| Light Mode | Dark Mode |
| ----------- | ----------- |
| <video src="https://github.com/wordpress-mobile/WordPress-iOS/assets/9609223/c55be6dd-4db5-4ca6-bbb9-85b1de58549e"></video> | <img width=350 src="https://github.com/wordpress-mobile/WordPress-iOS/assets/9609223/31180183-8df4-4e17-87db-58a6144aaad9" /> |

## Test Instructions

1. Install Jetpack and Login.
2. Enable `Domain Management` feature flag.
3. Head to Me > All Domains > Tap "+"
4. **Expect** the "Register Domain" to be presented modally with **Transfer Domain** view at the bottom.
5. Tap "Transfer Domain".
6. **Expect** the "Transfer Domain" web view to be presented modally.
7. Dismiss the web view.
8. Search for a domain then select a row.
9. **Expect** the "Transfer Domain" footer view to disappear and "Select Domain" to appear.
10. Start a new search.
11. **Expect** the "Transfer Domain" footer view to appear again.

## Regression Notes

1. Potential unintended areas of impact
Smoke test "Register Domain" screen.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

3. What automated tests I added (or what prevented me from doing so)
None.

# PR submission checklist:


# UI Changes testing checklist:
